app/api/ada/daedalusTransfer.js

```diff
@@ -82,15 +90,22 @@ export async function generateTransferTx(payload: {
     addressKeys,
     senderUtxos,
     outputAddr,
-  });
+  }, filterSender);
 }
 
-/** Generate transaction including all addresses with no change */
-export async function buildTransferTx(payload: {
-  addressKeys: AddressKeyMap,
-  senderUtxos: Array<UTXO>,
-  outputAddr: string,
-}): Promise<TransferTx> {
+/**
+  Generate transaction including all addresses with no change.
+  If filterSender is true, only include addresses that have outstanding
+  UTXOs in the senders property.
+*/
+export async function buildTransferTx(
+  payload: {
+    addressKeys: AddressKeyMap,
+    senderUtxos: Array<UTXO>,
+    outputAddr: string,
+  },
+  filterSenders : boolean = false
```

Comment:
Why does this need to have a default value and why is it not in `payload`? It's slightly more code change but much clear to put it with everything else I think.

Reply:
This defaults to false to keep the original semantics unchanged -- when used by Daedalus transfer, the `addressKeys` only contains addresses with outstanding UTXOs so filtering is not necessary. I will move it into `payload` though.

Reply:
BTW what's the point of grouping all parameters into `payload`?